pkg/apis/storage/validation/validation.go

```diff
@@ -403,7 +403,7 @@ func ValidateCSIDriver(csiDriver *storage.CSIDriver) field.ErrorList {
 
 // ValidateCSIDriverUpdate validates a CSIDriver.
 func ValidateCSIDriverUpdate(new, old *storage.CSIDriver) field.ErrorList {
-	allErrs := ValidateCSIDriver(new)
+	allErrs := apivalidation.ValidateObjectMetaUpdate(&new.ObjectMeta, &old.ObjectMeta, field.NewPath("metadata"))
```

Comment:
I would expect ValidateCSIDriver to call ValidateObjectMeta instead of ValidateCSIDriverName, just like ValidateCSINode:

```
allErrs := apivalidation.ValidateObjectMeta(&csiNode.ObjectMeta, false, apivalidation.ValidateNodeName, field.NewPath("metadata"))
```

etc

Reply:
@smarterclayton ,

I've spoken with @msau42 and @gnufied about this, and we agree that this is something that should be fixed; however, as a subsequent PR and not part of this one. If we make the change now it could break an existing workflow.

If we don't handle this change as part of the current PR, should we also revert the `ValidateCSIDriverUpdate` change from your prior comment?

Reply:
so it is an oversight that we did not use `ValidateObjectMeta` originally here and in `ValidateCSIDriver` function and as a result most metaobject validation is skipped. Now adding them here will result in tightening the validation as I understand it. That is why I am thinking we should fix this bit more carefully. 

Obviously it is a mess that we skipped those validations in the first place.. :(

Reply:
Yeah, we're completely broken in these objects and that's a big issue.  I don't know that we can ship 1.20 without some sort of fix here, but I don't think it has to be fixed in this PR exactly.

So let's get the important-critical issue opened (CSIDriver doesn't validate metadata changes), and then we can focus this PR on the minimum step.

Adding validate object metadata on create is safe, and we *must* validate object metadta before safe, and we *must* validate that name can't change.  So I would probably say this PR should add the validations and the issue is that before 1.20 we *must* deal with users with invalid data (that's a release blocker)

Reply:
https://github.com/kubernetes/kubernetes/issues/96520

Reply:
Given jordan's comments there, I think we are covered enough to add the validation and name validation on new and on update in this PR.